OpenRA.Game/Selection.cs

```diff
@@ -57,9 +57,10 @@ public void Combine(World world, IEnumerable<Actor> newSelection, bool isCombine
 				}
 			}
 
-			var voicedUnit = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && a.HasVoices());
-			if (voicedUnit != null)
-				Sound.PlayVoice("Select", voicedUnit, voicedUnit.Owner.Country.Race);
+			var voicedActor = actors.FirstOrDefault(a => a.Owner == world.LocalPlayer && a.IsInWorld && (a.TraitsImplementing<IVoiced>() != null));
```

Comment:
`.TraitsImplementing()` should never be able to return `null`. The proper check here is probably `.TraitsImplementing().Any()`.
But I suggest using `.HasTrait<IVoiced>()`, which is much simpler ;)